### PR TITLE
Show info in the item popup for things that expire at the end of a season

### PR DESCRIPTION
**Pull request by bhollis**
Screen Shot 2021-02-13 at 3 41 04 PM

The data is dirty, so I had to mess with it a bit, but I think this'll mostly work.

While I was at it, I decided to also show quest step:

Screen Shot 2021-02-13 at 4 05 09 PM

### CODE REVIEW

File: src/app/inventory/store/d2-item-factory.ts [hunk 1]
```diff
@@ -597,13 +597,30 @@ export function makeItem(
...
     createdItem.basePower = createdItem.primStat.value;
   }
 
   createdItem.index = createItemIndex(createdItem);
 
+  // Some items have multiple tooltips, but the item.tooltipNotificationIndexes property that
+  // should tell us which to show is missing: https://github.com/Bungie-net/api/issues/1419
+  if (
+    itemDef.tooltipNotifications?.length === 1 &&
+    itemDef.tooltipNotifications[0].displayString.length
```
Comment (kyleshay) [inline src/app/inventory/store/d2-item-factory.ts:610]: could this work in this case? optional chaining works on array indices as well`itemDef.tooltipNotifications.?[0].displayString.length`

**Review by kyleshay — APPROVED** [review]
nice one & the quest steps # are a good touch

File: src/app/inventory/store/d2-item-factory.ts (hunk 1 above)
Comment (kyleshay) [reply src/app/inventory/store/d2-item-factory.ts:610]: ah, hmm unless you're checking for the length to specifically be "`1`"?

Comment (bhollis) [reply src/app/inventory/store/d2-item-factory.ts:610]: I am looking for it to be 1.

**Pull request merged**